sdk/core/core-rest-pipeline/src/policies/exponentialRetryPolicy.ts

```diff
@@ -70,7 +70,12 @@ export function exponentialRetryPolicy(
    * @param retryData -  The retry data.
    * @returns True if the operation qualifies for a retry; false otherwise.
    */
-  function shouldRetry(statusCode: number | undefined, retryData: RetryData): boolean {
+  function shouldRetry(response: PipelineResponse | undefined, retryData: RetryData): boolean {
+    const statusCode = response?.status;
+    if (statusCode && statusCode === 503 && response && response.headers.get("Retry-After")) {
```

Comment:
I think this works and looks nicer?

```suggestion
    if (statusCode === 503 && response?.headers.get("Retry-After")) {
```